Makefile

```diff
@@ -2,15 +2,25 @@
 # Configuration Variables #
 ###########################
 # Image URL to use all building/pushing image targets
-export IMAGE_REPO ?= quay.io/operator-framework/operator-controller
-export IMAGE_TAG ?= devel
-export CERT_MGR_VERSION ?= v1.9.0
-export CATALOGD_VERSION ?= $(shell go list -mod=mod -m -f "{{.Version}}" github.com/operator-framework/catalogd)
-export KAPP_VERSION ?= $(shell go list -mod=mod -m -f "{{.Version}}" github.com/vmware-tanzu/carvel-kapp-controller)
-export RUKPAK_VERSION=$(shell go list -mod=mod -m -f "{{.Version}}" github.com/operator-framework/rukpak)
-export WAIT_TIMEOUT ?= 60s
-IMG?=$(IMAGE_REPO):$(IMAGE_TAG)
-TESTDATA_DIR := testdata
+ifeq (, $(IMAGE_REPO))
+IMAGE_REPO := quay.io/operator-framework/operator-controller
+endif
+export IMAGE_REPO
+
+ifeq (, $(IMAGE_TAG))
+IMAGE_TAG := devel
+endif
+export IMAGE_TAG
+
+IMG := $(IMAGE_REPO):$(IMAGE_TAG)
+
+
+# Define dependency versions (use go.mod if we also use Go code from dependency)
+export CERT_MGR_VERSION := v1.9.0
+export CATALOGD_VERSION := $(shell go list -mod=mod -m -f "{{.Version}}" github.com/operator-framework/catalogd)
```

Comment:
How does this new variable assignment impact doing something like:
```sh
make run CATALOGD_VERSION=vX.Y.Z
```

IIUC using the `?=` variable assignment would only expand the variables if they were not already defined via something like the above example. Is overriding the values still possible with this new variable assignment?

Reply:
If we want the variables to be overridable, they would need this sort of construction:
```
ifeq (,$(VAR))
VAR := $(shell echo "Hello")
endif
```

Does the catalogd version need to be configurable like that? If so, I can change it. Let me know if there are other variables in the same boat that you see with the non-overridable assignment.